### PR TITLE
refactor(NodeOutputFileSystem): upgrade to ES6

### PR DESCRIPTION
**Pull request by willmendesneto**
**What kind of change does this PR introduce?**

ES5 => ES6 refactor

**Did you add tests for your changes?**

refactor, all existing tests pass

**If relevant, link to documentation update:**

N/A

**Summary**

Helping in part of the ES6 refactor

**Does this PR introduce a breaking change?**

Nope

### CODE REVIEW

File: lib/node/NodeOutputFileSystem.js [hunk 1]
```diff
@@ -2,16 +2,21 @@
...
-function NodeOutputFileSystem() {}
-module.exports = NodeOutputFileSystem;
+const fs = require("fs");
+const path = require("path");
+const mkdirp = require("mkdirp");
+
+class NodeOutputFileSystem {
+	constructor() {
+		this.mkdirp = mkdirp;
+		this.mkdir = fs.mkdir.bind(this);
```
Comment (sokra) [inline lib/node/NodeOutputFileSystem.js]: This should bind `fs` instead of this, shouldn't it?

Comment (willmendesneto) [inline lib/node/NodeOutputFileSystem.js]: This part I just keep how it was. I did this change and the tests are passing, but is that ok do this change?

Comment (willmendesneto) [inline lib/node/NodeOutputFileSystem.js]: cc/ @sokra Can you answer this question, please? https://github.com/webpack/webpack/pull/3901#discussion_r96791351

**Comment by willmendesneto** [comment]
@sokra I updated with the changes. Can you take a look again, please?

**Comment by sokra** [comment]
Thanks

**Pull request merged**